### DISCUSSION TITLE
Can beets Return Tags For a File/Collection at Given Path?

### DISCUSSION THREAD

**Discussion by env-media** [beetbox/beets#4725] [Q&A]
I am writing a `bash` script to nicely re-import and update the music database whenever I move my library to a different location or the like. I am running into some edge cases where the artist name either begins or ends with a special character. Currently my script assumes the last component of the path is the artist's `albumartist` field, but it would be really nice if I could rely on beets to give me that information when provided a file or directory `path` instead. 

My questions are:
Does this functionality exist? If not, could I add in this functionality? If so, could you point me where in the code base I should look at adding this in?

**Comment by DjSlash** [comment]
Why are you writing a script for something that [beets already can do](https://beets.readthedocs.io/en/stable/faq.html#point-beets-at-a-new-music-directory)? Or is there something I missed here?

**Reply by env-media** [reply]
You missed the fact that I am reimporting collections/items after transferring to a new location or device entirely. My script resolves discrepancies between the music database and what exists on the file system.

**Reply by arsaboo** [reply]
So it is not clear from your original post what you are looking for. Are you looking for the artist information from beets?

**Reply by env-media** [reply]
Sorry for the lack of clarity.

Yup. I would like to issue a command to beets like `beets fetch '$albumartist' <path>` or something to that effect that will spit out the field of the passed in file/directory.

**Reply by arsaboo** [reply]
You can use something like `beet ls -f '$albumartist' album:"Jersey" year:2022 | head -n 1` to get the `$albumartist`
